adding new navigation to and refactoring SIS views

drewvolz opened the pull request:
[view without whitespace](https://github.com/StoDevX/AAO-React-Native/pull/6087/files?w=1)

This is the last of the big efforts to bring navigation back to the main views.


Todo: 
- #3986

--- 
While I’ve marked this PR as a draft, there’s a bit more to figure out with types.  I also don’t know how we can even test things out without a login. But if the changes are too much to go through I will go and break this up into smaller pull requests.

drewvolz committed:
add scrollview to noticeview in place of view

drewvolz committed:
add icon support to loading and notice views

drewvolz committed:
add warning icon to menu's noticeview in the error screen

drewvolz committed:
add download icon to course search's first-time loading screen

drewvolz committed:
allow overriding of icon color

drewvolz committed:
cleanup balances-acknowledgments

drewvolz committed:
cleanup balances

drewvolz committed:
cleanup and rename android student work detail

drewvolz committed:
cleanup and rename ios student work detail

drewvolz committed:
cleanup studentwork index

drewvolz committed:
export student work detail views and navigation options

drewvolz committed:
clean up course search

drewvolz committed:
clean up course search results

drewvolz committed:
clean up course search list

drewvolz committed:
clean up course search detail

drewvolz committed:
type cleanup for course search recent list

drewvolz committed:
export course search views and navigation options

drewvolz committed:
add new navigator to sis index

drewvolz committed:
clean up tes view

drewvolz committed:
update navigation types and routes for sis, jobs, course search

drewvolz committed:
remove unused view and colors from course search results

drewvolz committed:
update counts, problems, and errors

hawkrives reviewed (commented):
👍

drewvolz committed:
wrap student work data fetching in try/catch and report error values

drewvolz committed:
wrap sis balance fetching in try/catch and report error values

drewvolz committed:
update error counts and problems

drewvolz committed:
update tsc-errors

hawkrives reviewed (approved):
Rubber stamped - I don't have the mental bandwidth to review for a few weeks.

drewvolz committed:
replace deprecated Clipboard imports in call-phone and send-email

drewvolz committed:
make use of useDebounce hook to ease up on course result searching

drewvolz committed:
fix style type in course search result

drewvolz committed:
allow recent search and filters to be copied to the clipboard

pressable's onLongPress gives us a way to provide a different gesture
on the recents-list items.

this commit also takes the variable form of setting the pushbuttoncell
styles from above and uses it in-place of redefining them in this style

drewvolz committed:
replace animated searchbar with native searchbar in recent course search

* remove old searchbar
* add new searchbar using navigation.getParent() due to tab navigator
* use lodash's debounce to perform queries after a 1.5 second timeout
* add contentInsetAdjustmentBehavior for the searchbar/content layout

drewvolz committed:
update counts, errors, and problems

drewvolz commented on source/views/sis/course-search/search.tsx:
```diff
@@ -36,87 +33,110 @@ type ReduxDispatchProps = {
+				},
+			},
+		})
+	}, [navigation, typedQuery])
```
After testing this I am seeing UI inconsistencies. The searchbar is added to other views in the tab navigator which is not correct.

This could be happening because of how the tab navigator is nested. I am going through the documentation of how to set this up correctly.

- This in particular: https://reactnavigation.org/docs/screen-options-resolution/
- Hints from someone's wiki: https://github.com/evangreer91/Dev-Notes/wiki/Nesting-navigators
- Helpful reference: https://reactnavigation.org/docs/tab-based-navigation

hawkrives replied:
Or is it time to break SIS up into multiple top-level buttons?

Tab | Button
---|---
Balances | OleCard
TES | TES
Course Search | SIS
Student Work | … Student Work?

drewvolz replied:
Sorry, I meant to say this is **definitely** happening due to `navigation.getParent()?.setOptions()` -- the `.getParent` part being key.

This is our first tab navigation with top-level search in a tab. It works just fine if it's one-level deep, though... meaning Home > SIS tabs > Search > Results, where results has a searchbar.

Breaking these up is still an option, though. We could keep the grouping of things on the home screen if that helps visually separate them from the rest, so like a SectionList of tiled buttons on Home?

drewvolz replied:
I am happy to explore other UI designs, too. How do we feel about a list view?

I think there is a concern about breaking muscle memory on the home screen so if we limited our exploration to the subviews, that could be beneficial?

<img src="https://user-images.githubusercontent.com/5240843/171053399-306e7486-0222-4da3-9791-0e8eeaca8845.png" width=350 />

drewvolz replied:
Moved this out into https://github.com/StoDevX/AAO-React-Native/issues/6109 in the interest of solving this in a follow-up.

drewvolz committed:
revert noticeview scrollview and icon changes

drewvolz committed:
Merge branch 'master' into drew/sis

drewvolz committed:
undo getParent addition to the searchbar hook

drewvolz committed:
revert TES noticeview changes

drewvolz committed:
update errors and counts
